src/ledger/test/InMemoryLedgerTxn.h

```diff
@@ -5,8 +5,8 @@
 // of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
 
 #include "database/Database.h"
-#include "ledger/InMemoryLedgerTxnRoot.h"
 #include "ledger/LedgerTxn.h"
+#include "ledger/test/InMemoryLedgerTxnRoot.h"
 
 // This is a (very small) extension of LedgerTxn to help implement in-memory
```

Comment:
I think this comment could do with an update that clarifies:

  - In-memory mode was _originally_ for production uses but is now _just_ a helper mechanism for old tests that committed directly to the SQL database (bypassing the BL), that emulates the existence of a full SQL database, and that if we had the energy to change all those tests we could theoretically remove it. This is a potential future project! Maybe even file a (low priority for now) bug about it.
  - It is not entirely in-memory, but also writes-through to the offers table in SQL. This is surprising and contradicts the existing comment.

Reply:
Also I think you can ifdef-guard the whole thing on BUILD_TESTS?

Reply:
I don't think an ifdef is required given that I moved all the in memory mode files to the test directory and guard non-test file references to the in-memory mode with ifdefs.